From 5bf8b1d75750f6a621339a379d74f811dd37068b Mon Sep 17 00:00:00 2001
From: Paul Moore <pmoore@redhat.com>
Date: Wed, 3 Sep 2014 10:51:59 -0400
Subject: [PATCH 522/634] selinux: fix a problem with IPv6 traffic denials in
 selinux_ip_postroute()

commit a7a91a1928fe69cc98814cb746d5171ae14d757e upstream.

A previous commit c0828e50485932b7e019df377a6b0a8d1ebd3080 ("selinux:
process labeled IPsec TCP SYN-ACK packets properly in
selinux_ip_postroute()") mistakenly left out a 'break' from a switch
statement which caused problems with IPv6 traffic.

Thanks to Florian Westphal for reporting and debugging the issue.

Change-Id: I37a695b2ae0e474052c250f411e3da271eaa6f6c
Reported-by: Florian Westphal <fwestpha@redhat.com>
Signed-off-by: Paul Moore <pmoore@redhat.com>
---
 security/selinux/hooks.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/security/selinux/hooks.c b/security/selinux/hooks.c
index 1d386f4..1eaa51e 100644
--- a/security/selinux/hooks.c
+++ b/security/selinux/hooks.c
@@ -5005,6 +5005,7 @@ static unsigned int selinux_ip_postroute(struct sk_buff *skb, int ifindex,
 			case PF_INET6:
 				if (IP6CB(skb)->flags & IP6SKB_XFRM_TRANSFORMED)
 					return NF_ACCEPT;
+				break;
 			default:
 				return NF_DROP_ERR(-ECONNREFUSED);
 			}
-- 
1.9.1

